README.md

```diff
@@ -349,14 +367,18 @@ If you're importing a module that uses CommonJS you need to import as
 import * as _ from 'lodash';
 ```
 
-yarn lockfile
+Managing Dependencies (via yarn)
 -------------
 
-This project makes use of yarn to ensure that the exact same dependency versions are used every time you install it.
+This project makes use of [`yarn`](https://yarnpkg.com/en/) to ensure that the exact same dependency versions are used every time you install it.
 
-yarn creates the file [`yarn.lock`](https://yarnpkg.com/en/docs/yarn-lock) to track those versions. That file is updated automatically every time you install a new dependency from the commandline (by using `yarn add some-lib --save` or `yarn add some-lib --save-dev`).
+* `yarn` creates a [`yarn.lock`](https://yarnpkg.com/en/docs/yarn-lock) to track those versions. That file is updated automatically by whenever dependencies are added/updated/removed via yarn.
+* **Adding new dependencies**: To install/add a new dependency (third party library), use [`yarn add`](https://yarnpkg.com/en/docs/cli/add). For example: `yarn add some-lib --save`.
+    * If it's a development-only dependency (not for production), use `yarn add some-lib --save-dev`
```

Comment:
> If it's a development-only dependency (not for production)

This is more about build tools vs dependencies, rather than development vs production. Dependencies that are only needed to build the project like a typescript compiler, or webpack (an analogy with dspace's java code would be maven or ant) are dev dependencies, while things that need to end up in the built code, like angular itself, or bootstrap (for java: apache.commons, or solr) are regular dependencies.

Reply:
Thanks for the notes, @artlowel .  I've updated the README to clarify, and also rebased on `master` (so that the Travis build errors go away)